test/e2e/framework/job/wait.go

```diff
@@ -68,6 +70,22 @@ func WaitForJobComplete(ctx context.Context, c clientset.Interface, ns, jobName
 	})
 }
 
+// WaitForJobSuspend uses c to wait for suspend condition for the Job jobName in namespace ns.
+func WaitForJobSuspend(ctx context.Context, c clientset.Interface, ns, jobName string) {
+	gomega.Eventually(ctx, func() error {
```

Comment:
Here you are using Go code to check one attribute of a job. If this fails, the `job should be suspended` error becomes the failure reported by `gomega.Eventually`, with no additional information about what the actual status of the job is.

A better approach is to repeatedly get the job, then use gomega to check for the expected status. See https://github.com/kubernetes/kubernetes/blob/dc8b57d8a75764e649d641ac77c1539a574d9aed/test/e2e/framework/pod/wait.go#L221-L239 for an example.

Reply:
But here we are not waiting for the job to exist (we rather know it does). We are waiting for it to be picked up by the reconciler.

Can we use `EventuallyWithOffset`?

Reply:
> But here we are not waiting for the job to exist (we rather know it does). We are waiting for it to be picked up by the reconciler.

So that's what the custom matcher must check for.

If it's known that it exists, then the `framework.RetryNotFound` should not be used. Then `NotFound` will be treated as an error which aborts polling immediately.

> Can we use EventuallyWithOffset?

And use which offset? When using 1, the assumption is that `WaitForJobSuspend` is always called directly by tests. That's not necessarily true: it might also be called by some helper function in a test.

Reply:
My suggestion is to do something like the `framework.Gomega` in `WaitForPodCondition` also in `WaitForJobSuspend`.

Reply:
I updated the PR